### PR TITLE
Fixed moving from PC to MTP (phone) storage

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Fixes an issue preventing move operation to succeed on MTP (phone) devices

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files/Filesystem/StorageItems/SystemStorageFile.cs [hunk 1]
```diff
@@ -77,22 +77,48 @@ public override IAsyncAction CopyAndReplaceAsync(IStorageFile fileToReplace)
...
         public override IAsyncAction MoveAsync(IStorageFolder destinationFolder, string desiredNewName)
         {
-            return File.MoveAsync(destinationFolder, desiredNewName);
+            return MoveAsync(destinationFolder, desiredNewName, NameCollisionOption.FailIfExists);
         }
 
         public override IAsyncAction MoveAsync(IStorageFolder destinationFolder, string desiredNewName, NameCollisionOption option)
         {
-            return File.MoveAsync(destinationFolder, desiredNewName, option);
+            return AsyncInfo.Run(async (cancellationToken) =>
```
Comment (d2dyno1) [inline src/Files/Filesystem/StorageItems/SystemStorageFile.cs:90]: Why not Task.Run?

Comment (gave92) [reply src/Files/Filesystem/StorageItems/SystemStorageFile.cs:90]: Need to return an AsyncAction. That's how the IStorageItem interface is.

File: src/Files/Filesystem/StorageItems/SystemStorageFile.cs [hunk 2]
```diff
@@ -77,22 +77,48 @@ public override IAsyncAction CopyAndReplaceAsync(IStorageFile fileToReplace)
...
+                // Move unsupported, copy but do not delete original
+            });
         }
 
         public override IAsyncAction MoveAndReplaceAsync(IStorageFile fileToReplace)
         {
-            return File.MoveAndReplaceAsync(fileToReplace);
+            return AsyncInfo.Run(async (cancellationToken) =>
+            {
+                using (var inStream = await this.OpenStreamForReadAsync())
```
Comment (d2dyno1) [inline src/Files/Filesystem/StorageItems/SystemStorageFile.cs:114]: Is it possible to set the FileShare.Read?

Comment (gave92) [reply src/Files/Filesystem/StorageItems/SystemStorageFile.cs:114]: Not explicitly, but I've verified that copying a file that is open works correctly.

**Review by d2dyno1 — APPROVED** [review]

**Pull request merged**